components/alpaca/alpaca.app.mjs

```diff
@@ -1,11 +1,158 @@
+import { axios } from "@pipedream/platform";
+//Did not use Alpaca JS SDK because it does not work with OAuth
 export default {
   type: "app",
   app: "alpaca",
-  propDefinitions: {},
+  propDefinitions: {
+    isPaperAPI: {
+      type: "boolean",
+      label: "Paper API",
+      description: "Set to `true` if Paper API is used. Default is `false`.",
+      optional: true,
+    },
+    cancelOrders: {
+      type: "boolean",
+      label: "Cancel Orders",
+      description: "If `true` is specified, cancel all open orders before liquidating all positions.",
+      optional: true,
+    },
+    side: {
+      type: "string",
+      label: "Side",
+      description: "`buy` or `sell`",
+      options: [
+        "buy",
+        "sell",
+      ],
+    },
+    orderId: {
+      type: "string",
+      label: "Order Id",
+      description: "Order Id",
+      async options({
+        prevContext,
+        isPaperAPI,
+      }) {
+        const pageSize = 25;
+        const after = prevContext.nextAfter;
+        const orders = await this.getOrders({
+          isPaperAPI,
+          params: {
+            limit: pageSize,
+            after,
+            direction: "asc",
+          },
+        });
+        const nextAfter = orders[orders.length - 1]?.updated_at;
```

Comment:
In the last page when `nextAfter` becomes `undefined`, do the options start again from the beginning?

Reply:
@andrewjschuang you are right I guess I forgot to add the `if` statement there